[Schema Processor] Part 1 - Updating internal package from schema to translation

MovieStoreGuy opened the pull request:
**Description:**
As the schema processor improves, this package path changes to help make
sense of what is trying to do.

**Link to tracking Issue:** <Issue number if applicable>
I forget the actual link to the issue  but I shall find it

**Testing:**

Added unit tests for changes

**Documentation:** 

No behaviour / user facing changes happening, so ignored.

tigrannajaryan commented on processor/schemaprocessor/internal/translation/version.go:
```diff
@@ -0,0 +1,169 @@
+}
+
+// ReadVersionFromPath allows for parsing paths
+// that end in a schema indentifier
```
```suggestion
// that end in a schema version number string
```

tigrannajaryan commented on processor/schemaprocessor/internal/translation/version.go:
```diff
@@ -0,0 +1,169 @@
+	if strings.HasSuffix(p, "/") {
+		return nil, fmt.Errorf("must not have trailing slash: %w", ErrInvalidVersion)
+	}
+	ident := path.Base(p)
```
Nit:
```suggestion
	verNumStr := path.Base(p)
```

tigrannajaryan commented on processor/schemaprocessor/internal/translation/version.go:
```diff
@@ -0,0 +1,169 @@
+}
+
+// GetFamilyAndVersion takes a schemaURL and separates the family from the identifier.
+func GetFamilyAndVersion(schemaURL string) (family string, id *Version, err error) {
```
```suggestion
func GetFamilyAndVersion(schemaURL string) (family string, version *Version, err error) {
```

tigrannajaryan commented on processor/schemaprocessor/internal/translation/version.go:
```diff
@@ -0,0 +1,169 @@
+	return u.String(), id, err
+}
+
+func joinSchemaFamilyAndVersion(family string, version *Version) string {
```
This is unused so unclear how it will be used. Also not quite clear what it actually does. Does it only use the "path" option of the URL? If so the func name seems not completely accurate. Perhaps delete from this PR and  introduce when it is needed.

MovieStoreGuy replied:
This is used in a future PR but I forgot to clean it up when I split up the big complete implementation.

tigrannajaryan reviewed (commented):
Mostly LGTM, just some minor comments.

MovieStoreGuy committed:
[Schema Processor] Updating internal package from schema to translation

As the schema processor improves, this package path changes to help make
sense of what is trying to do.

MovieStoreGuy committed:
Addressing comments

MovieStoreGuy committed:
Adding in changelog

MovieStoreGuy committed:
Updating sum

tigrannajaryan commented:
@open-telemetry/collector-contrib-approvers please review.

tigrannajaryan commented:
@MovieStoreGuy please rebase the subsequent PRs and ping me when the next one is ready for review.
